hedera-node/test-clients/src/main/java/com/hedera/services/bdd/junit/support/translators/ConsensusSubmitMessageTranslator.java

```diff
@@ -0,0 +1,56 @@
+/*
+ * Copyright (C) 2024 Hedera Hashgraph, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hedera.services.bdd.junit.support.translators;
+
+import com.hedera.hapi.block.stream.output.StateChange;
+import com.hedera.hapi.block.stream.output.StateChanges;
+import com.hedera.hapi.node.transaction.TransactionReceipt;
+import com.hedera.hapi.node.transaction.TransactionRecord;
+import com.hedera.node.app.state.SingleTransactionRecord;
+import edu.umd.cs.findbugs.annotations.Nullable;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class ConsensusSubmitMessageTranslator implements TransactionRecordTranslator<SingleTransactionBlockItems> {
+    @Override
+    public SingleTransactionRecord translate(
+            @NotNull SingleTransactionBlockItems transaction, @Nullable StateChanges stateChanges) {
+        final var receiptBuilder = TransactionReceipt.newBuilder();
+
+        final var txnOutputItem = transaction.output();
+        if (txnOutputItem != null && txnOutputItem.hasSubmitMessage()) {
+            final var submitMessageOutput = txnOutputItem.submitMessage();
+            final var version = submitMessageOutput.topicRunningHashVersion().protoOrdinal();
+            receiptBuilder.topicRunningHashVersion(version);
+        }
+
+        stateChanges.stateChanges().stream()
+                .filter(StateChange::hasMapUpdate)
+                .findFirst()
+                .ifPresent(stateChange -> {
+                    final var topic = stateChange.mapUpdate().value().topicValue();
+                    receiptBuilder.topicSequenceNumber(topic.sequenceNumber());
+                    receiptBuilder.topicRunningHash(topic.runningHash());
+                });
+
+        return new SingleTransactionRecord(
+                transaction.txn(),
+                TransactionRecord.newBuilder().receipt(receiptBuilder.build()).build(),
+                List.of(),
+                new SingleTransactionRecord.TransactionOutputs(null));
```

Comment:
Since we're not sure how to construct sidecars, maybe we create a dummy object that does nothing, instead of using `List.of()`. Just a thought. Otherwise, this class makes a lot of sense, well done!

Reply:
@mhess-swl 
On this topic, can't we use sidecars populated for certain transaction from [BlockStreamBuilder](https://github.com/hashgraph/hedera-services/blob/1033baa9f6feec07fd58d6c95ff3ea24decb3055/hedera-node/hedera-app/src/main/java/com/hedera/node/app/blocks/impl/BlockStreamBuilder.java#L267-L293)? Later, we can access them by `transaction.output().contractCreate().sidecars()` for example

Reply:
I'm not very familiar with sidecar records, so maybe they will come from `output()`. Maybe they also come from state changes. Sounds like a good question to ask.